aptos-move/block-executor/src/executor.rs

```diff
@@ -409,13 +413,109 @@ where
         &self,
         txn_idx: TxnIndex,
         versioned_cache: &MVHashMap<T::Key, T::Tag, T::Value, X, T::Identifier>,
+        scheduler: &Scheduler,
+        shared_counter: &AtomicU32,
         last_input_output: &TxnLastInputOutput<T, E::Output, E::Error>,
         base_view: &S,
     ) {
-        let delta_keys = last_input_output.delta_keys(txn_idx);
-        let _events = last_input_output.events(txn_idx);
-        let mut delta_writes = Vec::with_capacity(delta_keys.len());
-        for k in delta_keys.into_iter() {
+        println!("worker commit hook called");
+        let parallel_state = ParallelState::<T, X>::new(versioned_cache, scheduler, shared_counter);
+        let latest_view = LatestView::new(base_view, ViewState::Sync(parallel_state), txn_idx);
+
+        // For each aggregator v2 in resource write set, replace the identifiers with values.
+        let mut write_set_keys = vec![];
+        let resource_write_set = last_input_output.resource_write_set(txn_idx);
+        let mut patched_resource_write_set = HashMap::new();
+        if let Some(resource_write_set) = resource_write_set {
+            for (key, (write_op, layout)) in resource_write_set.iter() {
+                // layout is Some(_) if it contains an aggregator
+                if layout.is_some() && !write_op.is_deletion() {
+                    write_set_keys.push(key.clone());
+                    let patched_bytes = match latest_view.replace_identifiers_with_values(
+                        write_op.bytes().unwrap(),
+                        layout.as_ref().unwrap(),
+                    ) {
+                        Ok((bytes, _)) => bytes,
+                        Err(_) => unreachable!("Failed to replace identifiers with values"),
+                    };
+                    let mut patched_write_op = write_op.clone();
+                    patched_write_op.set_bytes(patched_bytes);
+                    patched_resource_write_set.insert(key.clone(), patched_write_op);
+                }
+            }
+        }
+
+        // For each resource that satisfies the following conditions,
+        //     1. Resource is in read set
+        //     2. Resource is not in write set
+        // replace the aggregator v2 identifiers in the resource with corresponding values.
+        // If any of the aggregator v2 identifiers in the resource are part of aggregator_v2_write_set,
+        // then include the resource in the write set.
+        let aggregator_v2_keys = last_input_output.aggregator_v2_keys(txn_idx);
+        let read_set = last_input_output.read_set(txn_idx);
+        // TODO: Gives some error without this `mut`. See if we can avoid `mut` here.
+        if let Some(mut aggregator_v2_keys) = aggregator_v2_keys {
+            if let Some(read_set) = read_set {
+                for (key, data_read) in read_set.as_ref().data_reads.iter() {
```

Comment:
instead of making data_reads public, can we have a function that returns iterator over (&V, &MoveTypeLayout) - i.e. get_versioned_read_values() or something